src/test/regress/expected/with_dml.out

```diff
@@ -106,6 +106,10 @@ DEBUG:  Plan 16 query after replacing subqueries and CTEs: SELECT tenant_id FROM
 -- not a very meaningful query
 -- but has two modifying CTEs along with another
 -- modify statement
+-- We need to force 1 connection per placement
+-- otherwise the coordinator insert select fails
+-- since COPY cannot be executed
+SET citus.force_max_query_parallelization TO on;
```

Comment:
Note that this is a regression that the new executor leads to. I added an item to the main issue #2665:

```
The unified executor may regress for certain types of distributed queries. 
The most notable ones is that modifying ctes followed by an coordinator insert/select. The 
reason is that the previous modifications might use the same connections for modifying
 the same placements. However, the final coordinator INSERT/SELECT would require 1
 connection per placement. We should probably automatically detect these cases and 
force 1 connection per placement in advance.
```
However, forcing here to (a) make more regression tests pass (b) test the new GUC with a real use-case for it.

Let me know if this doesn't look good to you.